hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
Could you send the same figures with the max histogram so that we can compare? Also, are there spikes in the data?